### ISSUE TITLE
Network/AccessPoint - does not detect password requirement

### ISSUE THREAD

**Issue by regenman** [linkfrg/ignis#83]
### Already reported ? *

- [X] I have searched the existing open and closed issues.

### Regression?

No

### System Information

 ```
$ ignis systeminfo
Ignis 0.3.dev0
Branch: main
Commit: 756ee855443e07eb29c2a451e2dcc33ed16d08eb (fix!: use Utils.DebounceTask.run() for running the task instead of __call__; allow passing *args and **kwargs to the Utils.debounce decorator)
Current desktop: niri

os-release:
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo
```

### Description

This is possibly not an Ignis issue(?), but still: I don't get the WifiConnectDialog when trying to connect to any wifi network that needs a password. This is on NetworkManager (1.50.0) running with iwd (3.2) as backend. 

I narrowed it down to the NM_80211ApFlags not being seen through the PyGObject API:

```
~ $ python
Python 3.12.7 (main, Oct  1 2024, 11:15:50) [GCC 14.2.1 20240910] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import gi
>>> gi.require_version('NM', '1.0')
>>> from gi.repository import NM
>>> client=NM.Client.new(None)
>>> for dev in client.get_devices():
...     if dev.get_device_type() == NM.DeviceType.WIFI:
...             wifidev = dev
...
>>> for ap in wifidev.get_access_points(): print(f"{ap.get_ssid().get_data().decode()[::2]}  :  {ap.props.flags}")
...
CmeAB0  :  <flags 0 of type NM.80211ApFlags>
CME udeg  :  <flags 0 of type NM.80211ApFlags>
CmeAB0-G  :  <flags 0 of type NM.80211ApFlags>
CME_313  :  <flags 0 of type NM.80211ApFlags>
Tl230a  :  <flags 0 of type NM.80211ApFlags>
ZxlC4  :  <flags 0 of type NM.80211ApFlags>
Tl2150  :  <flags 0 of type NM.80211ApFlags>
Nrhl  :  <flags 0 of type NM.80211ApFlags>
CME_6da  :  <flags 0 of type NM.80211ApFlags>
Amrmr1 u.,  :  <flags 0 of type NM.80211ApFlags>
CME udeg5  :  <flags 0 of type NM.80211ApFlags>
CME_65a5h  :  <flags 0 of type NM.80211ApFlags>
CME_3135  :  <flags 0 of type NM.80211ApFlags>
cme_6D75  :  <flags 0 of type NM.80211ApFlags>
Bu  :  <flags 0 of type NM.80211ApFlags>
H-rn-DOfcjtPo63  :  <flags 0 of type NM.80211ApFlags>
CME_65a  :  <flags 0 of type NM.80211ApFlags>
cme_6D7  :  <flags 0 of type NM.80211ApFlags>
CME_f22  :  <flags 0 of type NM.80211ApFlags>
Tl2423  :  <flags 0 of type NM.80211ApFlags>
Tl2374  :  <flags 0 of type NM.80211ApFlags>
```
(I'm surrounded by password-protected networks, so there should be other flags than 0 on these APs. Note, SSID:s are obfuscated a bit for privacy.)
Of course, no flags reported results in the authentication dialogue not being triggered by Ignis. 

`nmcli dev wifi` correctly shows "WPA2" under the SECURITY column of these networks, and tools like nmtui or nm-applet can correctly detect password protected networks and prompt accordingly, so it seems that it's just the GObject API not reporting correctly, or is there another property we could/should query?

Investigating further, I found that there is indeed a property that correctly predicts password requirement on the AP: 
```

>>> for ap in wifidev.get_access_points(): print(f"{ap.get_ssid().get_data().decode()[::2]}  :  {ap.props.rsn_flags}")
...
CmeAB0  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
CME udeg  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
CmeAB0-G  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
CME_313  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
Tl230a  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
ZxlC4  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
Tl2150  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
Nrhl  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
CME_6da  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
Amrmr1 u.,  :  <flags 0 of type NM.80211ApSecurityFlags>
CME udeg5  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
CME_65a5h  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
CME_3135  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
cme_6D75  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
Bu  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
H-rn-DOfcjtPo63  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
CME_65a  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
cme_6D7  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
CME_f22  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
Tl2423  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
Tl2374  :  <flags NM_802_11_AP_SEC_PAIR_CCMP | NM_802_11_AP_SEC_GROUP_CCMP | NM_802_11_AP_SEC_KEY_MGMT_PSK of type NM.80211ApSecurityFlags>
>>>
```
Note that the only network with 0 flags here ("Amrmr1 u."), is indeed not password protected. 

In summary - I'm not sure if this is Ignis' fault or actually a deficiency in NetworkManager (possibly in relation to iwd), or if it's by design not reporting NM.80211ApFlags in this particular configuration, but it seems that a workaround for me would be to query the RSN flags (NM.80211ApSecurityFlags) instead. 

Finally, I'm seeing some interesting messages from NetworkManager with `nmcli general logging level TRACE`:
```
dec 21 17:53:28 NetworkManager[325021]: <debug> [1734800008.0175] device[6c136a585ea0b6e9] (wlan0): BSSID / quality / rate platform query failed
dec 21 17:53:40 NetworkManager[325021]: <debug> [1734800020.0226] device[6c136a585ea0b6e9] (wlan0): BSSID / quality / rate platform query failed
```

Thanks for any insights. 

### How to reproduce

See issue description.

**Comment by linkfrg** [comment]
Probably yes, we need to use ``rsn_flags`` and ``wpa_flags`` in ``requires_password`` property to check if an access point uses WPA/WPA2/WPA3 security.

I will try this after #80 to avoid possible merge conflicts

**Comment by linkfrg** [comment]
try #86

**Comment by regenman** [comment]
Cool, it seems to detect password requirement correctly now, but there's a leftover reference to `requires_password` in `access_point.py`:
```
  File "/usr/lib/python3.13/site-packages/ignis/services/network/access_point.py", line 203, in connect_to
    if self.requires_password:
       └ <access_point.WifiAccessPoint object at 0x7c1009395740 (ignis+services+network+access_point+WifiAccessPoint at 0x64699f9d2de0)>
  File "/usr/lib/python3.13/site-packages/ignis/gobject.py", line 166, in __getattribute__
    return super().__getattribute__(name)
                                    └ 'requires_password'

AttributeError: 'WifiAccessPoint' object has no attribute 'requires_password'
```

**Comment by regenman** [comment]
(also, the quick settings wouldn't show at all, until I added a call to `update_box` in `quick_settings.py` - I suspect it's due to changes in #80)

```
def quick_settings() -> Widget.Box:
    main_box = Widget.Box(vertical=True, css_classes=["qs-main-box"])
    update_box(main_box)  ##   <<<--- Added this line
    network.wifi.connect("notify::devices", lambda x, y: update_box(main_box))
    network.ethernet.connect("notify::devices", lambda x, y: update_box(main_box))
    network.vpn.connect("notify::connections", lambda x, y: update_box(main_box))

    return main_box
```

**Comment by linkfrg** [comment]
> Cool, it seems to detect password requirement correctly now, but there's a leftover reference to `requires_password` in `access_point.py`:
> 
> ```
>   File "/usr/lib/python3.13/site-packages/ignis/services/network/access_point.py", line 203, in connect_to
>     if self.requires_password:
>        └ <access_point.WifiAccessPoint object at 0x7c1009395740 (ignis+services+network+access_point+WifiAccessPoint at 0x64699f9d2de0)>
>   File "/usr/lib/python3.13/site-packages/ignis/gobject.py", line 166, in __getattribute__
>     return super().__getattribute__(name)
>                                     └ 'requires_password'
> 
> AttributeError: 'WifiAccessPoint' object has no attribute 'requires_password'
> ```

fixed in 27878f4127f711c293c3ce9a58547b124d9ef38e

**Comment by linkfrg** [comment]
> (also, the quick settings wouldn't show at all, until I added a call to `update_box` in `quick_settings.py` - I suspect it's due to changes in #80)
> 
> ```
> def quick_settings() -> Widget.Box:
>     main_box = Widget.Box(vertical=True, css_classes=["qs-main-box"])
>     update_box(main_box)  ##   <<<--- Added this line
>     network.wifi.connect("notify::devices", lambda x, y: update_box(main_box))
>     network.ethernet.connect("notify::devices", lambda x, y: update_box(main_box))
>     network.vpn.connect("notify::connections", lambda x, y: update_box(main_box))
> 
>     return main_box
> ```

It fixed in linfkrg/dotfiles#53, I will push temporary fix to the main soon

**Comment by regenman** [comment]
Thank you! It works as expected now. Feel free to close the issue.

**Comment by linkfrg** [comment]
> EDIT: Hmm, upon subsequent attempts to connect to a different network, there's another error thrown:
> 
> ```
> File "/home/ns/.config/ignis/modules/control_center/wifi.py", line 14, in <lambda>
>     on_click=lambda x: access_point.connect_to_graphical(),
>                     │  │            └ <function WifiAccessPoint.connect_to_graphical at 0x73b255e420c0>
>                     │  └ <access_point.WifiAccessPoint object at 0x73b2569aca40 (ignis+services+network+access_point+WifiAccessPoint at 0x59ff84daf980)>
>                     └ <wifi.WifiNetworkItem object at 0x73b2541df300 (modules+control_center+wifi+WifiNetworkItem at 0x59ff85e5dee0)>
> 
>   File "/usr/lib/python3.13/site-packages/ignis/services/network/access_point.py", line 248, in connect_to_graphical
>     WifiConnectDialog(self)
>     │                 └ <access_point.WifiAccessPoint object at 0x73b2569aca40 (ignis+services+network+access_point+WifiAccessPoint at 0x59ff84daf980)>
>     └ <class 'ignis.services.network.wifi_connect_dialog.WifiConnectDialog'>
>   File "/usr/lib/python3.13/site-packages/ignis/services/network/wifi_connect_dialog.py", line 14, in __init__
>     super().__init__(
>   File "/usr/lib/python3.13/site-packages/ignis/widgets/regular_window.py", line 33, in __init__
>     app.add_window(namespace, self)
>     │   │          │          └ <wifi_connect_dialog.WifiConnectDialog object at 0x73b23297d900 (ignis+services+network+wifi_connect_dialog+WifiConnectDialog...
>     │   │          └ 'wifi_connect'
>     │   └ <function IgnisApp.add_window at 0x73b2575dfd80>
>     └ <app.IgnisApp object at 0x73b2575ff580 (ignis+app+IgnisApp at 0x59ff8429a2c0)>
>   File "/usr/lib/python3.13/site-packages/ignis/app.py", line 410, in add_window
>     raise WindowAddedError(window_name)
>           │                └ 'wifi_connect'
>           └ <class 'ignis.exceptions.WindowAddedError'>
> 
> ignis.exceptions.WindowAddedError: Window already added: "wifi_connect"
> ```

This error is not related to the current issue and worth opening a new one
I probably know the root of the problem and the possible fix

**Comment by regenman** [comment]
Ok, so we can close this then and I'll open a new issue.
